lib/cylc/cfgspec/globalcfg.py

```diff
@@ -176,6 +176,15 @@ def upg( cfg, descr ):
     u.upgrade()
     u.deprecate( '6.1.2', ['task messaging', 'connection timeout in seconds'], ['task messaging', 'connection timeout'] )
     u.deprecate( '6.1.2', ['task messaging', 'retry interval in seconds'], ['task messaging', 'retry interval'] )
+    for batch_sys_name in ['loadleveler', 'lsf', 'pbs', 'sge', 'slurm']:
+        u.deprecate('6.4.1',
+            ['test battery', 'directives', batch_sys_name + ' host'],
+            ['test battery', 'batch systems', batch_sys_name, 'host'])
+        u.deprecate('6.4.1',
+            ['test battery', 'directives', batch_sys_name + ' directives',
+             '__MANY__'],
+            ['test battery', 'batch systems', batch_sys_name, 'directives',
+             '__MANY__'])
```

Comment:
The call to ```upgrade()``` should come last as it actions the declared deprecations - I see we already have this wrong since 6.1.2.  Also, my tests indicate the parsec upgrade code isn't working in this case...